src/lib/rpmostree-package.c

```diff
@@ -154,7 +154,7 @@ rpm_ostree_package_cmp (RpmOstreePackage *p1, RpmOstreePackage *p2)
    * when we read it out of the commit metadata and we also sort
    * the diff in_rpm_ostree_diff_package_lists().
    **/
-  ret = rpmvercmp (p1->evr, p2->evr);
+  ret = rpmverCmp (rpmverParse(p1->evr), rpmverParse(p2->evr));
```

Comment:
It looks like `rpmverParse` allocates and must be freed with `rpmverFree()`.

Reply:
See https://github.com/rpm-software-management/rpm/blob/35739c2a2298e61caacb45157706bf342ffcd20e/rpmio/rpmlua.c#L428

Reply:
good catch, will update!

Reply:
I didn't know this offhand; but as a very general rule in C (and C++ mostly) it's really rare that any function named something like "parse" returns a constant value, so I went to look.